sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala

```diff
@@ -109,6 +112,11 @@ case class GetCurrentDatabaseAndCatalog(catalogManager: CatalogManager) extends
         Literal.create(currentNamespace, StringType)
       case CurrentCatalog() =>
         Literal.create(currentCatalog, StringType)
```

Comment:
like this, we should pre-calculate the current user before transforming the query plan.